src/DocumentSession.h

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2021, Kung-Yao Hsieh, Shang-Chen LIN, Yi-Hang Xie. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef HTTPLIB_SRC_DOCUMENTSESSION_H
+#define HTTPLIB_SRC_DOCUMENTSESSION_H
+
+#include <string>
+#include <memory>
+#include <mutex>
+#include <set>
+
+class DocumentManager;
+class SocketSession;
+
+/** @brief Manage the operation between SocketSession and DocumentManager. */
+class DocumentSession {
+private:
+    /** @brief Store DocumentManager pointer. It is used to synchronize datas. */
+    DocumentManager* doc_manager_;
+    /** @brief Store SocketSession object pointer. */
+    std::set<SocketSession*> sessions_;
```

Comment:
You'll need comparison operator for set.

Reply:
Pointer type has comparison operator.